bio_nautilus.py

```diff
@@ -0,0 +1,294 @@
+from abc import ABC, abstractmethod
+from Bio import SeqIO
+from Bio.SeqUtils import gc_fraction
+
+
+class BiologicalSequence(ABC):
+    """Abstract base class representing a biological sequence."""
+
+    @abstractmethod
+    def __len__(self):
+        """Return the length of the sequence."""
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index):
+        """
+        Get a single character at index or a subsequence via slicing.
+
+        Args:
+            index: Integer position or slice object.
+
+        Returns:
+            Single character (str) for index, or subsequence (BiologicalSequence) for slice.
+        """
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        """Return the sequence as a plain string."""
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        """Return an unambiguous string representation of the object."""
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        """Check if all characters in the sequence are valid."""
+        pass
+
+
+class NucleicAcidSequence(BiologicalSequence, ABC):
+    """Abstract class representing a nucleic acid sequence (DNA/RNA)."""
+
+    _valid_chars = None
+    _complement_map = None
+
+    def __init__(self, sequence):
+        """
+        Initialize a nucleic acid sequence.
+
+        Args:
+            sequence: Input sequence string.
+
+        Raises:
+            ValueError: If invalid characters are present.
+            NotImplementedError: if NucleicAcidSequence is called directly.
+        """
+        if self.__class__ == NucleicAcidSequence:
+            raise NotImplementedError(
+                "NucleicAcidSequence является абстрактным классом и не может быть создан напрямую."
```

Comment:
```suggestion
                "NucleicAcidSequence является абстрактным классом и не может создавать экземпляров."
```

Но лучше конечно по английски